fedimint-server/src/config/api.rs

```diff
@@ -571,6 +714,20 @@ impl ConfigGenState {
 
         Ok(ConfigGenParams { local, consensus })
     }
+
+    fn reset(&mut self) {
+        self.config = None;
+        self.peers = Default::default();
+        self.auth = None;
+        self.requested_params = None;
+        self.status = ServerStatus::AwaitingPassword;
+        self.local = None;
+
+        info!(
+            target: fedimint_logging::LOG_NET_PEER_DKG,
+            "Reset config gen state"
+        );
+    }
```

Comment:
Do we need to reset any on-disk state? I think the `PASSWORD` file has been written already at that point?

Reply:
at this point, `PASSWORD` or any other config file is already written to a file in a temporary dir. `remove_staged_config()` resets disk state by deleting the whole dir, and this function finally resets server state